Bump Python version, assorted others

milljm opened the pull request:
Bump the version of latest Python allowed to 3.9.
Bump support libraries to their latest versions.

Closes #17105

Commit:
Remove the restriction on Python 3.8

Not really understanding the syntax. I have found issues with Python 3.8 and had
hoped to skip supporting that version all together... but I don't see how that is
possible given the following is being refused by conda:

  >=3.6,<3.8,>=3.9

Maybe order doesn't matter. Or there exists a logic operator (!) I am not familiar
with...

Refs #17105

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/690079/) on 3f9dc57 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/17106/site)

This comment will be updated on new commits.

Commit:
Update gfortran on Mac to 9.3.0

Commit:
Update compilers

Fix the conflict for OSX/Blas/LLVM

Commit:
Fix vkt/petsc version dependency on mpich build

Missed a couple things... MPICH gets rebuilt, everything must get rebuilt.

Commit:
Bump only gfortran version

Bumping blas versions, as those are required to be bumped when gfortran is bumped.

Commit:
Do not bump Fortran version

milljm commented:
@aeslaughter Seems to be VTK version issues causing the image diffs. I am looking into it. But unfortunately the way it stands right now, the only version of VTK available with Python 3.9, is 9.0.1.

aeslaughter commented:
I am alright with moving up to VTK9, do you need me to look into it?

milljm commented:
I think VTK 9 will work, so long as we re-gold things. But, there are other issues (as seen in the Mac Test failures).

Commit:
Bump Python version, assorted others

Bump the version of latest Python allowed to 3.9.
Bump support libraries to their latest versions.

Closes #17105

milljm commented:
I know there is a libMesh update in progress @lindsayad. These changes will certainly need to be included in your PR: https://github.com/idaholab/moose/pull/18268

Which ever goes in first I suppose?

moosebuild commented:
Job [Internal App tests](https://civet.inl.gov/job/771785/) on cf459dc : invalidated by @lindsayad

lindsayad commented:
sounds good. I suspect that your PR here will go in first, and then I'll rebase

milljm commented:
I have another commit that I want to include with this conda package update. Stand by.

Commit:
Syncronize tools/libmesh

Use a common conda_build_config.

Attempt to add dependencies not normally required by moose-tools,
but instead to possibly eleviate the solve time when folks install
moose-tools and moose-libmesh.

Updating blas requires every package to be updated.

milljm commented:
@lindsayad This is a hefty change. Let me know if you help updating libMesh.

I've basically consolidated the conda_build_config.yaml file, so stop forgetting to update multiple packages when necessary. And for sanity purposes of having it all in one place.

milljm commented on conda/mpich/conda_build_config.yaml:
```diff
@@ -1,15 +1,26 @@
-CONDA_BUILD_SYSROOT:           # [osx]
-  - /opt/MacOSX10.9.sdk        # [osx]
+#### The idea here is that anything below something else requires
+#### anything above that to be updated.
```
dumbest comment ever. I will change this...

lindsayad commented:
> This is a hefty change. Let me know if you need help updating libMesh.

will do!

Commit:
Specific moose-tools version(s)/arch

The extensive solve time is due to the fact we are not specifically setting versions.
Setting noarch, also played a factor in conflicts (so remove that).
